### PR TITLE
CI workflow to test deployment scripts

### PR DESCRIPTION
**Pull request by nkuba**
This workflow will run deployment scripts against the local hardhat network instance. This is to test the scripts.

Depends on https://github.com/keep-network/coverage-pools/pull/103

### CODE REVIEW

File: .github/workflows/contracts.yml [hunk 1]
```diff
@@ -34,6 +34,24 @@ jobs:
           FORKING_URL: ${{ secrets.MAINNET_ETH_HOSTNAME }}
         run: yarn test:system
 
+  contracts-deployment-test:
```
Comment (michalinacienciala) [inline .github/workflows/contracts.yml:37]:
```suggestion
  contracts-deployment:
```
Could we drop the `-tests` from the name? I don't see any tests performed in this job.

Comment (nkuba) [reply .github/workflows/contracts.yml:37]: It is a test run of the deployment scripts on local hardhat network instance. I know the name is confusing. Maybe I'll update it to `contracts-deployment-dry-run`?

Comment (michalinacienciala) [reply .github/workflows/contracts.yml:37]: Oh, I see. Im fine with your proposition of the change.

File: .github/workflows/contracts.yml [hunk 2]
```diff
@@ -34,6 +34,24 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "12.x"
+
+      - name: Cache yarn directory
+        uses: c-hive/gha-yarn-cache@v1
```
Comment (michalinacienciala) [inline .github/workflows/contracts.yml:47]: We could try using this functionality in `setup-node` action for caching:
```suggestion
      - uses: actions/setup-node@v2
        with:
          node-version: "12.x"
          cache: "yarn"
```
(Also applies to other jobs in the workflow)

**Review by michalinacienciala — APPROVED** [review]
Looks good to me, but I included one optional suggestion

File: .github/workflows/contracts.yml [hunk 3]
```diff
@@ -34,6 +35,25 @@ jobs:
...
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "12.x"
+          cache: "yarn"
+
+      - name: Cache yarn directory
+        uses: c-hive/gha-yarn-cache@v1
```
Comment (michalinacienciala) [inline .github/workflows/contracts.yml:49]: I think this duplicates the
```
      - uses: actions/setup-node@v2
        with:
          node-version: "12.x"
          cache: "yarn"
```
We can get rid of it (here and in other places in the workflow).

**Review by michalinacienciala — CHANGES_REQUESTED** [review]

**Review by michalinacienciala — APPROVED** [review]

**Pull request merged**